src/blog/2024/04/displaying-logged-in-users-on-dashboard.md

```diff
@@ -0,0 +1,148 @@
+
+**Option 1: Include client Data**
+
+By default, this option is enabled. When this option is enabled, an object with user information will be added to the “msg” emitted by any widget of the Node-red Dashboard 2.0.
+
+**Option 2: Accept client constraints**
+
+A feature that ensures messages are specifically targeted to individual clients, which enhances the precision and security of data transmission within the platform. It determines by enabling the nodes option in the FF Auth tab whether the enabled node type will utilize client data, such as socketid, and restrict communications to only that client.
+
+For example, consider a manufacturing facility where each production line has its own monitoring system. With this feature enabled, data from sensors on Production Line A will only be sent to the monitoring system designated for Production Line A. This ensures that data remains isolated and relevant to each specific area of operation, maintaining organizational efficiency and security.
+
+*Note: Please note that Multi-User Addons can only be used by our Teams and Enterprise Self-Hosted customers. Upon request, we provide all required configurations to get started.*
+
+!["Screenshot displaying the Flowfuse Muti-user addon option"](./images/displaying-logged-in-user-ff-auth-tab.png "Screenshot displaying the Flowfuse Muti-user addon option")
+
+## Displaying logged in user on Dashboard 2.0
+We are all set to display logged in user on Dashboard 2.0. You can confirm this by printing the msg object emitted by the Dashboard 2.0 widget, which should have attached user information. Use the debug node for this.
+
+To display user information on the dashboard we will use Vue’s [Teleport](https://dashboard.flowfuse.com/nodes/widgets/ui-template.html#teleports) feature to render content to a specific location in the DOM, we will display user information at the action bar’s right-hand side.
+
+1. Drag an ui-template widget onto the canvas.
+2. Click on that node, and select type as “Widget (Ui-Scoped)”. ( this allows us to render this ui-template at ui scoped which means I will not required to add separate ui-templates for different pages )
+3. Copy the below vue snippet and paste that into the ui-template.
+
+```
+<template>
+    <!-- Teleporting user info to #app-bar-actions, which is id of actions bars right corners area -->
+    <Teleport v-if="loaded" to="#app-bar-actions">
+        <div class="user-info">
+            <!-- Displaying user image -->
+            <img :src="setup.socketio.auth.user.image" />
+            <!-- Greeting the user -->
+            <span>Hi, {{ setup.socketio.auth.user.name }}</span>
+        </div>
+    </Teleport>
+</template>
+
+<script>
+    export default {
+        data() {
+            return {
+                // Flag to indicate if the component is loaded
+                loaded: false
+            };
+        },
+        mounted() {
+            // Component is mounted, set loaded to true
```

Comment:
Can we have a comment that describes when `mounted()` is called and thus why loaded is set to true right after?

Reply:
Okay, ZJ. I'll add a comment explaining when mounted() is called and why loaded is set to true right after.